packages/flutter_test/test/accessibility_test.dart

```diff
@@ -23,6 +23,59 @@ void main() {
       handle.dispose();
     });
 
+    testWidgets('Multiple text with same label', (WidgetTester tester) async {
+      final SemanticsHandle handle = tester.ensureSemantics();
+      await tester.pumpWidget(
+        _boilerplate(
```

Comment:
Based on my reading of this implementation, we'll check the contrast of each of these text pieces twice. If it fails, we'll also report it more than once, right?

Reply:
That is correct. An unintended side effect of doing the loop over all text elements with same text.

Reply:
That ... doesn't seem like a good side effect. 

If we're doing the semantic node based approach, then I think we should make an effort to determine what the right node <-> element relationship is. Perhaps by looking at screen locations?

If we're doing the element based approach, then we drop this but at the cost of checking text that is ignored by semantics.

This sort of feels like its in an odd in-between place and I'm not certain that the API is going to make sense when you look at it from the perspective of an end user

Reply:
That sounds a perfect optimization to the current brute force approach. 

While this PR doesn't land it when the suggestion is but it does ease the developer concerns in other domains. 
1. In GP3 out of the 1215 contrast issues identified by this test ~470 of them are not failures but the infra throws an Exception when there are more than 1 similar labels. This makes the current test to some degree not useful. 

2. We have a giant exception list to bypass this test and have no clue of the real failures. 

While this does print the error more than once but it gets the developer in the right direction of what exactly to fix.

Reply:
I'm not concerned about performance, I'm concerned about correctness.

Reply:
Can you give a counter example where this is incorrect?

Reply:
The test case you posted in potentially one example. Note that the test framework is not currently flagging that test case as a failure. Please ensure that it fails without the exclude semantics

Reply:
Updated.